src/graphql/utils/buildObject.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * A utility function for easily building an object from lists *without*
+ * tolerating duplicate keys. If any duplicate keys are seen an error will be
+ * thrown.
+ *
+ * This is useful for when we are building map objects as is often done for
+ * GraphQL configs.
+ *
+ * This is pretty much identical to the Lodash `_.fromPairs` function.
+ *
+ * ## Example:
+ * ```js
+ * const actual = buildObject(
+ *   [
+ *     ['a', 1],
+ *     ['b', 2],
+ *   ],
+ *   [
+ *     ['c', 3],
+ *   ]
+ * )
+ *
+ * const expected = {
+ *   a: 1,
+ *   b: 2,
+ *   c: 3,
+ * }
+ *
+ * assert.deepEqual(actual, expected)
+ * ```
+ */
+export default function buildObject <T>(
+  ...entriess: Array<Array<[string, T] | undefined | null | false>>
+): { [key: string]: T } {
+  const object = {}
+
+  entriess.forEach(entries => entries.forEach(entry => {
+    if (!entry)
+      return
+
+    const [key, value] = entry
+
+    if (object.hasOwnProperty(key))
+      throw new Error(`Naming conflict when building object. Cannot have two definitions for property '${key}'.`)
```

Comment:
So I’m less inclined to find a solution to tables with `before` and `foo_before` until it becomes a problem (assuming the table name is `foo`). `before_` would have documentation, of course so it really should not be hard to tell the difference (just open up the documentation browser in GraphiQL). If we were going to do `fooBefore`, I’d want to prefix all the names. I feel like by prefixing one and not others you give it some level of importance that just wasn’t there before, it stands out. An underscore suffix makes it different, but not that different. Make sense?

The more we talk about this, the more I think just moving it into `condition` now is the right move. It cleans up the argument list for connections (major win in my opinion and something that’s a breaking change), and allows us to punt this decision for later if it turns out we do want these conditions at the top level.

For procedures this only breaks for query Postgres functions that return a set and have a conflicting name. I can live with that erroring for now.